packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -475,6 +476,24 @@ async function handler(
     ) {
       throw new HttpError({ statusCode: 403, message: ErrorCode.CancelledBookingsCannotBeRescheduled });
     }
+    const availableSlots = await getAvailableSlots({
```

Comment:
Not sure if we should be calling getAvailableSlots here. Later on ensureAvailableUsers is also called and both use getUserAvailability internally. Can you check if there is any impact of the performance?

I think you should add guests to the users that we are passing to ensureAvailableUsers.

cc @CarinaWolli would love you thoughts on this as I am not completely sure 🙏

Reply:
I moved most of the logic to the `ensureAvailableUsers` function since they are similar.